### ISSUE TITLE
False "Redundant else" with const if condition

### ISSUE THREAD

**Issue by camsteffen** [intellij-rust/intellij-rust#6333]
## Environment

* **IntelliJ Rust plugin version:** 0.3.133.3426-202
* **Rust toolchain version:** rustc 1.47.0
* **IDE name and version:** IntelliJ IDEA Ultimate 2020.2
* **Operating system:** macOS 10.15.7 (19H2)

## Problem description

A false positive (or at least misleading) "Redundant \`else`" warning occurs for the following code.

```rust
const A: bool = true;
let _ = if A {  // should perhaps warn here, on the const condition
    "hi"
} else {  // not here
    todo!()
};
```

<img width="388" alt="image" src="https://user-images.githubusercontent.com/5565418/97787977-83277900-1b83-11eb-9694-89891efe09e4.png">
